### PR TITLE
JSON stringify objects

### PR DESCRIPTION
**Pull request by chriddyp**
Also replace the slow `.map` calls with faster `for` loops.

Tests coming up

### CODE REVIEW

**Comment by chriddyp** [comment]
This is to fix the `[object Object]` display in the grid
image

**Comment by chriddyp** [comment]
heads up @cldougl for your screenshots - the postgis cells are going to be actual objects instead of `[object Object]`

**Comment by chriddyp** [comment]
@alexandresobolevski - any problems that you can think of off the top of your head with this approach? I'll test against date objects and geojson objects

**Comment by alexandresobolevski** [comment]
I got hit by the redhsift dates that is all. Those date objects need to be usable by workspace.

PS did not know `map` calls are worth changing for `for` loops when increasing speed. Do you know form the top of your head what impact in could have?

**Comment by chriddyp** [comment]
> PS did not know map calls are worth changing for for loops when increasing speed. Do you know form the top of your head what impact in could have?

It's like 95% faster. However, this only makes a difference in areas where we're looping over thousands of elements

**Comment by chriddyp** [comment]
ok, I think this is good now. @alexandresobolevski - could you review?

**Comment by chriddyp** [comment]
Here is a screenshot inside the workspace
image

**Comment by chriddyp** [comment]
Before this PR, that query looked like:
image

File: backend/parse.js [hunk 1]
```diff
@@ -31,11 +31,28 @@ export function parseSQL(data) {
...
-        )
-    );
+    const rows = [];
+    for (let i = 0; i < data.length; i++) {
+        rows[i] = [];
+        for (let j = 0; j < columnnames.length; j++) {
+            let cell = data[i][columnnames[j]];
+            // Like PostGIS's GeoJSON or Dates
+            const cellType = type(cell);
+            if (contains(cellType, ['Object', 'Boolean'])) {
```
Comment (alexandresobolevski) [inline backend/parse.js]: I think redshift dates come out as `Object` types not as `Dates`, I can't connect to it right now to see if I am right. Need to check this to make sure and need to open up that redshift database.

File: test/backend/Datastores.spec.js [hunk 2]
```diff
@@ -38,6 +39,42 @@ describe('SQL - ', function () {
             done();
         }).catch(done);
     });
+
+    /*
+     * Unfortunately we don't have a PostGIS table that doesn't update
+     * with random data yet, so we have to skip this test on CI.
+     * It worked at a single point in time.
+     * TODO - Create a static PostGIS table for testing
```
Comment (alexandresobolevski) [inline test/backend/Datastores.spec.js]: Sure thing!

File: backend/parse.js (hunk 1 above)
Comment (chriddyp) [inline backend/parse.js]: I couldn't connect to it at the office either..

Comment (chriddyp) [inline backend/parse.js]: OK finally connected. I think the `aapl_2014` table is the only one with dates. Looks OK:
image

**Pull request merged**